OpenRA.Mods.Common/TraitsInterfaces.cs

```diff
@@ -145,7 +145,24 @@ public interface IProductionIconOverlay
 		bool IsOverlayActive(ActorInfo ai);
 	}
 
-	public interface INotifyTransform { void BeforeTransform(Actor self); void OnTransform(Actor self); void AfterTransform(Actor toActor); }
+	public interface INotifyTransform
+	{
+		void BeforeTransform(Actor self);
+		void OnTransform(Actor self);
+		void AfterTransform(Actor toActor);
+	}
+
+	public interface INotifyDeploy
+	{
+		void FinishedDeploy(Actor self);
+		void FinishedUndeploy(Actor self);
+	}
+
+	public interface INotifyDeployAnimation
```

Comment:
I have looked at `Sellable` and `Transforms`, and I'm really afraid of the cans of worms that might open. 
I don't mind trying, but please let me do that in its own PR afterwards, because the activity interaction of those two (especially `Sellable`) is just asking for regressions, and this PR is large enough as it is, in my opinion.